### PR TITLE
fixes #1590: Moved Settings menu to it's own component

### LINKED ISSUE #1590: Move settings menu into its own component
We currently have the settings menu fully contained inside of the [`Header`](https://github.com/mozilla/testpilot/blob/master/frontend/src/app/components/Header.js) component. We should break it out into its own component.

### PR DESCRIPTION
**Pull request by pmespresso**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "babelify": "^7.3.0",
     "browserify": "^11.0.1",
     "chai": "^3.5.0",
+    "chai-enzyme": "^0.6.1",
```
Comment (fzzzy) [inline package.json]: This isn't actually used, better remove it.

File: frontend/src/app/components/Header.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
 import RetireConfirmationDialog from './RetireConfirmationDialog';
+import Settings from './Settings';
 
 export default class Header extends React.Component {
 
   constructor(props) {
     super(props);
+
     this.closeTimer = null;
+    this.showSettingsMenu = this.showSettingsMenu.bind(this);
```
Comment (fzzzy) [inline frontend/src/app/components/Header.js]: I prefer to put the call to bind at the place where this.showSettingsMenu is being passed, but I know I am probably in the minority about that.

It just seems gross to me to have some functions on this that are bound, and some that are not.

File: frontend/test/app/components/Header-test.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { expect } from 'chai';
 import sinon from 'sinon';
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
```
Comment (pmespresso) [inline frontend/test/app/components/Header-test.js]: shallow never used, removing import.

File: package.json (hunk 1 above)
Comment (pmespresso) [reply package.json]: Got it

File: frontend/src/app/components/Header.js (hunk 2 above)
Comment (pmespresso) [reply frontend/src/app/components/Header.js]: I can make that fix if it's necessary.

Comment (pmespresso) [reply frontend/src/app/components/Header.js]: But perhaps I should leave it for the sake of consistency with other components?

**Comment by fzzzy** [comment]
Thanks for the pull request!

**Comment by pmespresso** [comment]
Awesome! Happy to contribute!

**Pull request merged**